server/src/main/java/org/opensearch/index/remote/RemoteStoreEnums.java

```diff
@@ -112,6 +118,40 @@ boolean requiresHashAlgorithm() {
             }
         };
 
+        private final int code;
+
+        PathType(int code) {
+            this.code = code;
+        }
+
+        public int getCode() {
+            return code;
+        }
+
+        private static final Map<Integer, PathType> CODE_TO_ENUM;
+
+        static {
+            PathType[] values = values();
+            Map<Integer, PathType> codeToStatus = new HashMap<>(values.length);
+            for (PathType value : values) {
+                int code = value.code;
+                if (codeToStatus.containsKey(code)) {
+                    throw new IllegalStateException(
+                        new ParameterizedMessage("{} has same code as {}", codeToStatus.get(code), value).getFormattedMessage()
+                    );
+                }
+                codeToStatus.put(code, value);
+            }
+            CODE_TO_ENUM = unmodifiableMap(codeToStatus);
+        }
+
+        /**
+         * Turn a status code into a {@link PathType}.
+         */
+        public static PathType fromCode(int code) {
+            return CODE_TO_ENUM.get(code);
+        }
+
```

Comment:
instead of `fromCode`, `toCode` , can we implement `writeTo(StreamOutput out)` and `readFrom(StreamInput in)` methods. We can refer to `AllocationStatus` for a sample implementation  ?

Reply:
We dont have StreamInput or StreamOutput here.